doc/Deploy.md

```diff
@@ -0,0 +1,214 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# DAPHNE Packaging, Distributed Deployment, and Management
+
+### Overview
+
+This file ([doc/Deploy.md](Deploy.md)) explains deployment of **DAPHNE runtime systems** on HPC (e.g. w/ SLURM), and highlights the excerpts from descriptions of functionalities in [deploy/](../deploy/) directory, mainly [deploy-distributed-on-slurm.sh](../deploy/deploy-distributed-on-slurm.sh):
+- compilation of the Singularity image,
+- compilation within the Singularity image, of the codes for `daphne` main target and worker target,
+- packaging compiled daphne targets,
+- packaging compiled daphne targets with user payload as a payload package,
+- uploading the payload package to an HPC platform,
+- obtaining the list of `PEERS`from SLURM,
+- executing daphne main and worker binaries on SLURM `PEERS`,
+- collection of logs from daphne execution, and
+- cleanup of worker environments and payload deployment.
+
+### Deployment Functionalities for SLURM
+
+DaphneDSL runs within Runtimes, which consist of main worker and other distributed workers connected to it.
+To establish such Runtime with connections, a deployment of the Runtimes is necessary. Such deployment is described below.
+Running on the DAPHNE Distributed Runtime does not require any changes to the DaphneDSL code, but just to prepare the environment of the Deployment Platform beforehand since `daphne` figures out automatically what to run in a distributed fashion.
+
+The [deploy-distributed-on-slurm.sh](../deploy/deploy-distributed-on-slurm.sh) packages and starts targets on a target HPC platform, and is tailored to the communication required with SLURM and the target HPC platform.
+While the packaging and transfer script [deployDistributed.sh](../deploy/deployDistributed.sh) already provides some functionality, the specific upgraded functionalities in the extended [deploy-distributed-on-slurm.sh](../deploy/deploy-distributed-on-slurm.sh) compared to [deployDistributed.sh](../deploy/deployDistributed.sh) are:
+- The building of the `daphne` main and worker targets to be later started on distributed nodes, can be run through a Singularity container. The Singularity container can be built on the utilized HPC. Otherwise, the function `deploy` in [deployDistributed.sh](../deploy/deployDistributed.sh) sends and builds executables on each node, which might cause overwrite if the workers use same mounted user storage (e.g. distributed storage attached as home directory).
+- The list of `PEERS` is not defined by the user but obtained from SLURM (in `deployDistributed.sh`, the user supplies `PEERS` as an argument).
+- Specifying `SLURM` running time for single DAPHNE main target duration is provided (with `RunOneRequest`).
+- Cleanup support is added.
+    
+### How to use DAPHNE Packaging, Distributed Deployment, and Management of Runtime Systems
```

Comment:
Thanks @aristotelis96. This is now addressed in the commit 3420a3e9c4469999c481767af57153135b8820ac.
The command is now mentioned in the section's title. When additional sections are added (PR on `deployDistributed.sh` or otherwise), they can be done in a similar fashion.